adding buoyancy production to nonlinear k-epsilon model Refs #31000

tanoret opened the pull request:
Refs #31000

tanoret committed:
adding buoyancy production to nonlinear k-epsilon model Refs #31000

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -66,13 +72,23 @@ INSFVTKEDSourceSink::INSFVTKEDSourceSink(const InputParameters & params)
   if (_dim >= 3 && !_w_var)
     paramError("w", "In three or more dimensions, the w velocity must be supplied!");
+
+  if(_temperature && _alpha == 0)
```
```suggestion
  if(_temperature && !_alpha)
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -66,13 +72,23 @@ INSFVTKEDSourceSink::INSFVTKEDSourceSink(const InputParameters & params)
+  if(_temperature && _alpha == 0)
+    paramError("alpha", "The thermal expansion coefficient should be >0.0 for themal bouyancy production correction.");
+
+  if(_temperature && !(_gravity))
```
```suggestion
  if(_temperature && !_gravity)
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -66,13 +72,23 @@ INSFVTKEDSourceSink::INSFVTKEDSourceSink(const InputParameters & params)
+    paramError("alpha", "The thermal expansion coefficient should be >0.0 for themal bouyancy production correction.");
+
+  if(_temperature && !(_gravity))
+    paramError("gravity", "Gravity should be provided when bouyancy corrections are active.");
```
We need this errors for the turbulent prandtl number too I think.

tanoret replied:
Pr_t is already defaulted to 0.9, which is what we used in most turbulent modeling cases. I think is alright to use the default.

grmnptr commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven/tests:
```diff
@@ -319,6 +319,25 @@
   []
+  [lid_driven_turb_buoyant]
+    requirement = 'The system shall be able to run fluid flow with k-epsilon turbulence model in enclosures with buoyant production, '
+    [result]
+      type = 'Exodiff'
+      input = 'lid-driven-turb-energy-wall-bouyancy-production.i'
+      exodiff = 'lid-driven-turb-energy-wall-bouyancy-production_out.e'
+      detail = 'and reach converged results with segregated solvers.'
+      abs_zero = 1e-6
+      heavy = true
+    []
```
I think it is enough to have the run part here as a regression test. The real runs will come to open pronghorn soon.

tanoret replied:
Thanks. Removed the results test

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -91,6 +91,33 @@ where:
+\end{equation}
+
+where $\vec{u}_{par}$ is the component of the velocity vector parallel to the gravity vector
+and $\vec{u}_{perp}$ is the perpendicular one.
```
```suggestion
and $\vec{u}_{perp}$ is the velocity perpendicular to the gravity vector.
```

grmnptr reviewed (changes requested):
Beautiful, just a few comments.
Who can do the technical review on this, @freiler ?

GiudGiud reviewed (commented):
I ll try this more

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3034198/), step Clang format on 4f8be75 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/31001/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/31001/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format f9ceeedf956be1b512e268b2966893df7334e140`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3034415/), step Docs: sync website on a65eb80 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31001/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3034464/), step Generate coverage on a65eb80 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>f9ceee</th>
      <th colspan=4>#31001 a65eb8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.28%</td>
      <td>85.28%</td>
      <td>-0.00%</td>
      <td>88.89%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>19226</td>
      <td>19260</td>
      <td>+34</td>
      <td>40</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3318</td>
      <td>3325</td>
      <td>+7</td>
      <td>5</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31001/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31001/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31001/coverage/xfem)

</details>

## Warnings

- `navier_stokes` new line coverage rate 88.89% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud reviewed (changes requested):
just some typos

for the review, we should be able to compare with and without this term in the cavity tomorrow

tanoret committed:
correcting sign of c3_eps Refs #31000

GiudGiud committed:
Typos and test formatting

Let's hold on here since the other PR is modifying the folder structure and would conflict

tanoret committed:
adding relaminarization for nonlinear kepsilon Refs #31000

tanoret committed:
wall distance aux kernels to compute the distance to the wall Refs #31000

tanoret committed:
deleting trailing white spaces Refs #31000

tanoret committed:
correcting compile issues in kEpsilonViscosityAux Refs #31000

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3413732/), step Clang format on 7662fa1 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/31001/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/31001/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b85414d92b7659931cc90106a67659da1e979214`

Commit:
solving clang issues in kEpsilonViscosity Refs #31000

moosebuild commented:
Job [Test](https://civet.inl.gov/job/3414468/), step Results summary on 4d9ee83 wanted to post the following:

## Framework test summary

Compared against c1e94d1 in job [civet.inl.gov/job/3413879](https://civet.inl.gov/job/3413879).

No change
## Modules test summary

Compared against c1e94d1 in job [civet.inl.gov/job/3413879](https://civet.inl.gov/job/3413879).

### Removed tests

| Test                                                                                                                   | Time (s)   |
|------------------------------------------------------------------------------------------------------------------------|------------|
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy/run`                     | 3.44       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_capped/run`                     | 2.92       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy_wall/run`                | 2.05       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_linear_wall/run`                | 1.94       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_std_wall_bulk/run`              | 1.92       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy_wall_linear/run`         | 1.84       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_no_ts_limiter/run`              | 1.84       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_inc_wall/run`                   | 1.83       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_no_wall/run`                    | 1.77       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_non_eq_bulk/run`                | 1.73       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_std_wall/run`                   | 1.71       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_non_eq_wall/run`                | 1.68       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_no_wall_log_first_layer/run`    | 1.50       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy_wall_log/run`            | 1.48       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_non_eq_bulk/result`             | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_no_wall/result`                 | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_capped/result`                  | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_std_wall/result`                | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_inc_wall/result`                | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_non_eq_wall/result`             | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_no_wall_log_first_layer/result` | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_linear_wall/result`             | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy_wall_log/result`         | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy_wall/result`             | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_no_ts_limiter/result`           | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_std_wall_bulk/result`           | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy/result`                  | SKIP       |
| `navier_stokes/test:finite_volume/ins/turbulence/lid-driven/segregated.lid_driven_turb_energy_wall_linear/result`      | SKIP       |

### No added tests

Commit:
solving issue with missing tests in ins/turbulence/lid-driven/segregated Ref #31000
